src/gradient8.c

```diff
@@ -0,0 +1,96 @@
+#define TOPOTOOLBOX_BUILD
+
+#include <math.h>
+#include <omp.h>
```

Comment:
As it stands, I am pretty sure that a user that doesn't have OpenMP installed correctly may run into errors during compilation. 

I [tried](https://github.com/wkearn/libtopotoolbox/actions/runs/10632149219) deleting all the OpenMP installation steps from our CI job, and it continues to work on Windows with MSVC and Linux with GCC, because those compilers have OpenMP built in, but on Linux with clang and macOS, it fails.

I would prefer if everything continued working without OpenMP support. The problematic thing seems to be including the OpenMP header. I think that we technically don't need the header because we only use the compiler pragmas and not any of the OpenMP API functions (such as `omp_get_num_threads`).

Indeed [if I delete this header include](https://github.com/wkearn/libtopotoolbox/actions/runs/10632183173), it seems like everything continues to work -- though I don't have any confirmation that this is really running on multiple threads.

If we do need this header, I think the best thing to do is to wrap it in an `#if #endif` to include it only if we can confirm that OpenMP is available. As I recommended elsewhere, we probably want to define a `TOPOTOOLBOX_OPENMP_VERSION` macro anyway, so we could do something like

```C
#if TOPOTOOLBOX_OPENMP_VERSION > 0
#include <omp.h>
#endif
```

Then in `include/topotoolbox.h` we need to do something like

```C
#ifndef TOPOTOOLBOX_OPENMP_VERSION
#define TOPOTOOLBOX_OPENMP_VERSION 0
#endif
```
which will set TOPOTOOLBOX_OPENMP_VERSION to 0 unless it is defined by the build system. Finally, we would add

```CMake
find_package(OpenMP)
if (OpenMP_C_FOUND)
  target_compile_options(topotoolbox PUBLIC -DTOPOTOOLBOX_OPENMP_VERSION=${OpenMP_C_VERSION_MAJOR}${OpenMP_C_VERSION_MINOR})
  target_link_libraries(topotoolbox PUBLIC OpenMP::OpenMP_C)
endif()
```

to `src/CMakeLists.txt` to set the version.

See https://github.com/wkearn/libtopotoolbox/commit/eab82eb243db21f80f1ed042a9bdd07c3a5eb335 for those changes. It does seem to work.